pyirf/scripts/make_DL3.py

```diff
@@ -0,0 +1,338 @@
+"""Script to produce DL3 data from DL2 data and a configuration file.
+
+Is it initially thought as a clean start based on old code for reproducing
+EventDisplay DL3 data based on the latest release of the GADF format.
+
+Todo:
+- make some config arguments also CLI ones like in ctapipe-stage1-process
+
+
+"""
+
+# =========================================================================
+#                            MODULE IMPORTS
+# =========================================================================
+
+# PYTHON STANDARD LIBRARY
+
+import argparse
+import os
+
+# THIRD-PARTY MODULES
+
+import numpy as np
+import astropy.units as u
+from astropy.table import Table
+from astropy.coordinates.angle_utilities import angular_separation
+from gammapy.spectrum import cosmic_ray_flux, CrabSpectrum  # UPDATE TO LATEST
+
+# THIS PACKAGE
+
+from pyirf.io.io import load_config, read_FITS
+from pyirf.perf import (
+    CutsOptimisation,
+    CutsDiagnostic,
+    CutsApplicator,
+    IrfMaker,
+    SensitivityMaker,
+)
+
+
+def main():
+
+    # =========================================================================
+    #                   READ INPUT FROM CLI AND CONFIGURATION FILE
+    # =========================================================================
+
+    # INPUT FROM CLI
+
+    parser = argparse.ArgumentParser(description="Produce DL3 data from DL2.")
+
+    parser.add_argument(
+        "--config_file",
+        type=str,
+        required=True,
+        help="A configuration file like pyirf/resources/performance.yaml .",
+    )
+
+    parser.add_argument(
+        "--obs_time",
+        type=str,
+        required=True,
+        help="An observation time written as (value.unit), e.g. '50.h'",
+    )
+
+    parser.add_argument(
+        "--pipeline",
+        type=str,
+        required=True,
+        help="Name of the pipeline that has produced the DL2 files.",
+    )
+
+    parser.add_argument(
+        "--debug", action="store_true", help="Print debugging information."
+    )
+
+    args = parser.parse_args()
+
+    # INPUT FROM THE CONFIGURATION FILE
+
+    cfg = load_config(args.config_file)
+
+    # Add obs. time to the configuration file
+    str_obs_time = args.obs_time.split(".")
+    cfg["analysis"]["obs_time"] = {
+        "value": float(str_obs_time[0]),
+        "unit": str(str_obs_time[-1]),
+    }
+
+    # Get input directory
+    indir = cfg["general"]["indir"]
+
+    # Get template of the input file(s)
+    template_input_file = cfg["general"]["template_input_file"]
+
+    # Get output directory
+    outdir = os.path.join(
+        cfg["general"]["outdir"],
+        "irf_{}_Time{}{}".format(  # name could change in future
```

Comment:
This comment seems rather arbitrary, of course this can change.

As a general comments, I would just let users declare the output file name.

Reply:
ok, I'll remove the comment

though, if we want them to be compatible it's better we define it - if people shares these files is better they have a unique template no? we can decide later, it is not urgent

Reply:
No, we should not rely on filenames for that. Rather we use the corresponding FITS header keywords.

E.g. `CREATOR`, `TELESCOP`, ...

Reply:
This are mostly required by the OGADF standard anyway

Reply:
this would have been something to add later to #6 :)